wrappers/yandexcloudkms/yandexcloudkms.go

```diff
@@ -0,0 +1,256 @@
+	if config == nil {
+		config = map[string]string{}
+	}
+
+	// Check and set KeyID
+	k.keyID = coalesce(os.Getenv(EnvYandexCloudKMSKeyID), config[CfgYandexCloudKMSKeyID])
+	if k.keyID == "" {
+		return nil, fmt.Errorf(
+			"neither '%s' environment variable nor '%s' config parameter is set",
+			EnvYandexCloudKMSKeyID, CfgYandexCloudKMSKeyID,
+		)
+	}
+
+	// Check and set k.client
+	if k.client == nil {
+		client, err := getYandexCloudKMSClient(
+			coalesce(os.Getenv(EnvYandexCloudOAuthToken), config[CfgYandexCloudOAuthToken]),
+			coalesce(os.Getenv(EnvYandexCloudServiceAccountKeyFile), config[CfgYandexCloudServiceAccountKeyFile]),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error initializing Yandex.Cloud KMS client: %w", err)
+		}
+
+		if err := k.setClient(client); err != nil {
+			return nil, fmt.Errorf("error setting Yandex.Cloud KMS client: %w", err)
+		}
+	}
+
+	// Map that holds non-sensitive configuration info
+	wrappingInfo := make(map[string]string)
+	wrappingInfo["kms_key_id"] = k.keyID
+
+	return wrappingInfo, nil
+}
+
+func (k *Wrapper) setClient(client kms.SymmetricCryptoServiceClient) error {
+	k.client = client
+
+	// Make sure all the required permissions are granted (also checks if key exists)
+	_, err := k.Encrypt(context.Background(), []byte("go-kms-wrapping-test"), nil)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to encrypt with Yandex.Cloud KMS key - ensure the key exists and permission to encrypt the key is granted: %w", err)
+	}
+
+	return nil
+}
+
+// Init is called during core.Initialize. No-op at the moment.
+func (k *Wrapper) Init(_ context.Context) error {
+	return nil
+}
+
+// Finalize is called during shutdown. This is a no-op since
+// Wrapper doesn't require any cleanup.
+func (k *Wrapper) Finalize(_ context.Context) error {
+	return nil
+}
+
+// Type returns the wrapping type for this particular Wrapper implementation
+func (k *Wrapper) Type() string {
+	return wrapping.YandexCloudKMS
+}
+
+// KeyID returns the last known key id
+func (k *Wrapper) KeyID() string {
+	return k.currentVersionID.Load().(string)
+}
+
+// HMACKeyID returns the last known HMAC key id
+func (k *Wrapper) HMACKeyID() string {
+	return ""
+}
+
+// Encrypt is used to encrypt the master key using Yandex.Cloud symmetric key.
+// This returns the ciphertext, and/or any errors from this
+// call. This should be called after the KMS client has been instantiated.
+func (k *Wrapper) Encrypt(_ context.Context, plaintext, aad []byte) (blob *wrapping.EncryptedBlobInfo, err error) {
+	if plaintext == nil {
+		return nil, fmt.Errorf("given plaintext for encryption is nil")
+	}
+
+	env, err := wrapping.NewEnvelope(nil).Encrypt(plaintext, aad)
+	if err != nil {
+		return nil, fmt.Errorf("error wrapping data: %w", err)
+	}
+
+	if k.client == nil {
+		return nil, fmt.Errorf("nil client")
+	}
+
+	encryptResponse, err := k.client.Encrypt(
+		context.Background(),
```

Comment:
Don't use background -- use the context passed into this function.

Reply:
Fixed